gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/rolesprovider/github/GithubTeamsUserRolesProvider.groovy

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.security.rolesprovider.github
+
+import com.netflix.spinnaker.gate.config.GitHubProperties
+import com.netflix.spinnaker.gate.security.rolesprovider.UserRolesProvider
+import com.netflix.spinnaker.gate.security.rolesprovider.github.client.GitHubMaster
+
+import com.netflix.spinnaker.gate.security.rolesprovider.github.client.model.Team
+import com.netflix.spinnaker.gate.security.rolesprovider.github.client.model.TeamMembership
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.InitializingBean
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import org.springframework.util.Assert
+import retrofit.RetrofitError
+import retrofit.client.Response
+
+@Slf4j
+@Component
+@ConditionalOnProperty(value = "auth.groupMembership.service", havingValue = "github")
+class GithubTeamsUserRolesProvider implements UserRolesProvider, InitializingBean {
```

Comment:
More or less, it would replace the `GitHubClient` and `BasicAuthRequestInterceptor`s. You would `getForObject` passing in your `Team` or `TeamMembership` class, depending on the call.

https://spring.io/blog/2009/03/27/rest-in-spring-3-resttemplate